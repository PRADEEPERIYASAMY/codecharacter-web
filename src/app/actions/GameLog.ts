import { action } from 'typesafe-actions';

export namespace GameLogActions {
  export enum Type {
    UPDATE_GAME_LOGS = 'UPDATE_GAME_LOGS',
  }

  export const updateGameLog = (
    player1DebugLog: string,
    player2DebugLog: string,
    gameLog: string,
  ) =>
    action(Type.UPDATE_GAME_LOGS, {
      gameLog,
      player1DebugLog,
      player2DebugLog,
    });
}
