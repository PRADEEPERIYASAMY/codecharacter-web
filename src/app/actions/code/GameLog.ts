import { action } from 'typesafe-actions';

export namespace GameLogActions {
  export enum Type {
    FETCH_GAME_LOG = 'FETCH_GAME_LOG',
    UPDATE_GAME_LOG = 'UPDATE_GAME_LOG',
  }

  export const fetchGameLog = (gameLog: string) =>
    action(Type.FETCH_GAME_LOG, {
      gameLog,
    });

  export const updateGameLog = (gameLog: string) =>
    action(Type.UPDATE_GAME_LOG, {
      gameLog,
    });
}
