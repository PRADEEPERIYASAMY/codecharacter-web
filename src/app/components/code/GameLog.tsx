import * as React from 'react';
// tslint:disable-next-line:import-name
import AceEditor from 'react-ace';

import 'brace/mode/c_cpp';
import 'brace/theme/terminal';

export class GameLog extends React.Component<GameLog.Props, {}> {
  constructor(props: GameLog.Props) {
    super(props);
  }

  public render() {
    const { gameLog } = this.props;
    const options = {
      enableBasicAutocompletion: false,
      enableLiveAutocompletion: false,
      enableSnippets: false,
      showLineNumbers: true,
      tabSize: 2,
    };

    return (
      <AceEditor
        mode="c_cpp"
        theme={'terminal'}
        name="game_log_div"
        fontSize={14}
        wrapEnabled={true}
        readOnly={true}
        showPrintMargin={false}
        showGutter={true}
        highlightActiveLine={false}
        setOptions={options}
        editorProps={{ $blockScrolling: true }}
        width={'100%'}
        height={'100%'}
        value={gameLog}
      />
    );
  }
}

export namespace GameLog {
  export interface StateProps {
    gameLog: string;
  }

  export interface DispatchProps {
    updateGameLog: (gameLog: string) => void;
  }

  export type Props = StateProps & DispatchProps;
}
