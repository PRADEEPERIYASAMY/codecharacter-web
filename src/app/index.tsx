import { Dashboard } from 'app/components/Dashboard';
import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router';

/* tslint:disable-next-line:variable-name */
export const App = hot(module)(() => (
  <Switch>
    <Route path="/" component={Dashboard} />
  </Switch>
));
