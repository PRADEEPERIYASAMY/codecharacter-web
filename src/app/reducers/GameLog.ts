import { GameLogActions } from 'app/actions';
import * as GameLogInterfaces from 'app/types/GameLog';

const gameLogStoreIntialState: GameLogInterfaces.GameLogStoreState = {
  gameLog: '/* Game logs */',
  player1DebugLog: '',
  player2DebugLog: '',
};

export const gameLogReducer = (
  state = gameLogStoreIntialState,
  action: GameLogInterfaces.GameLogStoreAction,
) => {
  switch (action.type) {
    case GameLogActions.Type.UPDATE_GAME_LOGS: {
      return {
        ...state,
        gameLog: action.payload.gameLog,
        player1DebugLog: action.payload.player1DebugLog,
        player2DebugLog: action.payload.player2DebugLog,
      };
    }
    case GameLogActions.Type.RESET_GAME_LOG_STATE: {
      return {
        ...gameLogStoreIntialState,
      };
    }
    default:
      return state;
  }
};
