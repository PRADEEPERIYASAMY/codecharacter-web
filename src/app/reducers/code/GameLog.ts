import { GameLogActions } from 'app/actions';
import * as GameLogInterfaces from 'app/types/GameLog';

const gameLogStoreIntialState: GameLogInterfaces.GameLogStoreState = {
  value: '/* Game logs */',
};

export const gameLogReducer = (
  state = gameLogStoreIntialState,
  action: GameLogInterfaces.GameLogStoreAction,
) => {
  switch (action.type) {
    case GameLogActions.Type.UPDATE_GAME_LOG: {
      return {
        ...state,
        logCode: action.payload.gameLog,
      };
    }
    default:
      return state;
  }
};
