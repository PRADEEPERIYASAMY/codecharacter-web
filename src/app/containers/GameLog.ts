import { GameLogActions } from 'app/actions';
import { GameLog } from 'app/components/GameLog';
import { RootState } from 'app/reducers';
import * as GameLogInterfaces from 'app/types/GameLog';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    gameLog: rootState.gameLog.gameLog,
    player1DebugLog: rootState.gameLog.player1DebugLog,
    player2DebugLog: rootState.gameLog.player2DebugLog,
  };
};

const gameLogContainer = connect<GameLogInterfaces.StateProps, GameLogInterfaces.DispatchProps, {}>(
  mapStateToProps,
  {
    updateGameLog: GameLogActions.updateGameLog,
  },
)(GameLog);

export default gameLogContainer;
