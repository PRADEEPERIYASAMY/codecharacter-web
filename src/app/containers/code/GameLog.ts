import { GameLogActions } from 'app/actions';
import { GameLog } from 'app/components/code/GameLog';
import { RootState } from 'app/reducers';
import { connect } from 'react-redux';

const mapStateToProps = (rootState: RootState) => {
  return {
    gameLog: rootState.log.gameLog,
  };
};

const gameLogContainer = connect<GameLog.StateProps, GameLog.DispatchProps, {}>(
  mapStateToProps,
  {
    updateGameLog: GameLogActions.updateGameLog,
  },
)(GameLog);

export default gameLogContainer;
