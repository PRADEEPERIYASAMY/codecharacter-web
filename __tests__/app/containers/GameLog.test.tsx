import GameLog from 'app/containers/code/GameLog';
import { configureStore } from 'app/store';
import { shallow } from 'enzyme';
import * as React from 'react';
// tslint:disable-next-line:import-name
// import AceEditor from 'react-ace';

describe('GameLog Container', () => {
  const store = configureStore();
  const wrapper = shallow(<GameLog />, {
    context: {
      store,
    },
  });
  it('Should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have prop', () => {
    expect(wrapper.props().gameLog).toMatch(store.getState().gameLog.gameLog);
  });
});
