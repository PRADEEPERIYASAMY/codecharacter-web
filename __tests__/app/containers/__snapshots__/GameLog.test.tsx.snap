// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameLog Container Should render 1`] = `
<GameLog
  gameLog="/* Game logs */"
  player1DebugLog=""
  player2DebugLog=""
  updateGameLog={[Function]}
/>
`;
